sourcegraph/cody
# feat: custom starter message configuration

RE: https://sourcegraph.slack.com/archives/C05AGQYD528/p1693983800785199

This PR adds a new configuration `cody.customer.starter`  to allow setting a custom starter message that will prepend chat requests to Cody:
- Configuration now has a `cody.customer.starter` string property.
- A new setting cody.custom.starter was added in vscode/package.json to allow configuring a custom message.
- getConfiguration() was updated to read the new customStarter config value.
- ConfigurationWithAccessToken gained a customStarter property.
- main.ts imports PromptMixin to set the custom starter message on extension start.

## Test plan



1. set `cody.customer.starter` in your user config, e.g. "call me jeff"
2. reload extension
3. submit a chat request
4. check the output channel to see your setting was added to the chat request

<img width="2005" alt="Screenshot 2023-09-06 at 9 03 38 PM" src="https://github.com/sourcegraph/cody/assets/68532117/4988f06c-0301-4332-812f-4ef92078c474">
